base_layer/core/src/chain_storage/target_difficulties.rs

```diff
@@ -0,0 +1,80 @@
+//  Copyright 2020, The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::{
+    blocks::BlockHeader,
+    consensus::ConsensusManager,
+    proof_of_work::{PowAlgorithm, TargetDifficultyWindow},
+};
+
+#[derive(Debug, Clone)]
+pub struct TargetDifficulties {
+    monero: TargetDifficultyWindow,
```

Comment:
I disagree:
1. A HashMap can have anything in it, so requires more code to deal with 
1. A HashMap is allocated on the heap and so requires allocation 
1. Needs to be initialized at runtime
1. This makes it impossible at compile time to not have target difficulties for all algos (required for header sync)
1. This means that the `get` function does not have to return an `Option` or `Result` which makes client code easier (if you think about it, why should getting a Pow target difficulty window be optional? We know the Pow algos at compile time)
1. The unimplemented will be removed when we remove blake (or we can implement it) - so it's not a concern.

Reply:
Yeah,  this is more of a style thing. I don't mind either. The advantage of having a hashmap is that you can do `TargetDifficulties[pow.algo]` instead of matches everywhere,  but  that can be implemented here with a single method that does the match. Since it's done this way already, it's probably fine